DevTools/HackStudio/Editor.cpp

```diff
@@ -462,6 +468,25 @@ void Editor::set_document(GUI::TextDocument& doc)
 {
     ASSERT(doc.is_code_document());
     GUI::TextEditor::set_document(doc);
+
+    CodeDocument& code_document = static_cast<CodeDocument&>(doc);
+    switch (code_document.language()) {
+    case Language::Cpp:
+        set_syntax_highlighter(make<GUI::CppSyntaxHighlighter>());
+        cpp_Language_server_connection().post_message(Messages::CppLanguageServer::FileOpened(code_document.file_path().string()));
```

Comment:
@itamar8910 
I didn't really have the time to look at these until today, but...this is going to cause a rather big amount of duplication (also that it forces HackStudio to know about the different LSPs' it connects to statically, not great)

Take a look at https://github.com/alimpfard/serenity/commit/8837d87080571e2534ef063c6aaa1a6125fd6bbd for a preview of the duplication.
These languages servers should probably:
- have a single IPC definition
- have a base class (at least on the HackStudio side

Reply:
Yeah I agree 100%, it wouldn't be good to have this much code duplication when using language servers.
Feel free to do this architecture change in your branch (or we could do it later which is also OK).